ipsi/src/ipsi.c

```diff
@@ -185,11 +185,47 @@ static int initConnectionName(char *appName,char *connectionType){
  *
  * @param[in] connectionType: Specify the role of the application either "Caller" or "Server"
  *
- * @return SUCCESS: The application as successfully registered itself with IPSI library
+ * @return SUCCESS: The application successfully registered itself with IPSI library
  * @return FAILURE: The application was unable to register with IPSI library , an appropriate error message will be shown on console
  **/
 int ipsiConnectionType(char *appName, char *connectionType){
 
+	if(!(initConnectionName(appName,connectionType))){
+		IPSI_LOG("[IPSI] Invalid Connection Type, Exiting...\n");
+		return FAILURE;
+	}
+
+	//Initialize DBUS Code
+
+	//dbus_bus_get() will connect and register the client with DBus Daemon.
+
+	conn = dbus_bus_get(DBUS_BUS_SESSION, &err);
+	if (dbus_error_is_set(&err)) {
+		IPSI_LOG("[IPSI] Connection Error (%s)\n", err.message);
+		dbus_error_free(&err);
+		return FAILURE;
+	}
+	if (NULL == conn) {
+		IPSI_LOG("[IPSI] Connection Null\n");
+		return FAILURE;
+	}
+
+	//dbus_bus_request_name() asks the DBus to assign the given name (connectionName) to the connection (conn).
```

Comment:
Remove this and any excessive comment from this function

Reply:
okay.